pkg/util/pullsecret/pullsecret.go

```diff
@@ -9,13 +9,33 @@ import (
 	"os"
 	"reflect"
 
+	v1 "k8s.io/api/core/v1"
+
 	"github.com/Azure/ARO-RP/pkg/api"
 )
 
 type pullSecret struct {
 	Auths map[string]map[string]interface{} `json:"auths,omitempty"`
 }
 
+type SerializedAuthMap struct {
+	Auths map[string]SerializedAuth `json:"auths,omitempty"`
+}
+
+type SerializedAuth struct {
+	Auth string `json:"auth"`
+}
```

Comment:
Well it can but it will not be much readable, this copies the structure of the encoded secret. I firmly believe using proper de-serialization makes code easier to understand.

Reply:
Readability in this case is bias. I didnt had any issues reading this before, so why changing? 
Or if you changing to full deserialized struct change it everywhere. Having almost same 2 structs do not help readability.